controllers/lectureController.js

```diff
@@ -0,0 +1,89 @@
+var formidable = require("formidable");
+const fs = require("fs");
+var dir = __dirname + '/../uploads/lectures/';
+const {
+    Lecture
+} = require("../models/lecture.model");
+
+// write code for lecture uploadto folder and save in database.
+
+exports.createLecture = async(req, res, next) => {
+    var form = new formidable.IncomingForm();
+    form.parse(req, (err, fields, files) => {
+        if (err) {
+            res.send(err.message);
+        }
+        var newpath = '';
+        if (files != '') {
+            if (!fs.existsSync(dir)) {
+                fs.mkdir(dir, {
+                    recursive: true
+                }, (err) => {
+                    res.send(err);
+                });
+            }
+            var oldpath = files.lecture_video.path;
+            var newpath = __dirname + '/../uploads/lectures/' + files.lecture_video.name;
+
+            fs.rename(oldpath, newpath, function(fileErr) {
+                if (fileErr) throw fileErr;
+            });
+        }
+
+        var insertData = {
+            course_id: fields.course_id,
+            title: fields.title,
+            description: fields.description,
+            lecture_video: newpath,
+            video_link: fields.video_link,
+            user_id: fields.user_id,
+        };
+        Lecture.insertMany([insertData], function(error, result) {
+            if (error) {
+                res.send(error)
```

Comment:
Again, we should replace res.send to res.json() at all places. It adds all the necessary headers for you.